taglocker_test.go

```diff
@@ -0,0 +1,162 @@
+package plc
+
+import (
+	"fmt"
+	"math/rand"
+	"strings"
+	"testing"
+)
+
+const alphaCharset = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+const numCharset = "0123456789"
+const alphanumCharset = alphaCharset + "_" + numCharset
+
+// NoOpReadWriter is a dummy ReadWriter interface that unconditionally succeeds.
+type NoOpReadWriter struct {
+	state uint32
+}
+
+func (norw NoOpReadWriter) ReadTag(name string, value interface{}) error {
+	garbage := norw.state
+	garbage++
```

Comment:
I think you just want `_ = norw.state`. Underscore just means "unused variable".